src/pages/Settings/components/PackageModalForm.tsx

```diff
@@ -122,7 +122,21 @@ export const SettingsPackageModalForm: React.FC<{
           </ProForm.Item>
         </>
       ) : (
-        <ProFormText width="lg" name="value" label={<FormattedMessage id="value" />} />
+        <>
+          <ProForm.Item shouldUpdate label={<FormattedMessage id="value" />}>
+            {() =>
+              form.getFieldValue('rules')?.includes('boolean') ? (
+                <ProFormSwitch
+                  name="value"
+                  checkedChildren={<FormattedMessage id={'true'} />}
+                  unCheckedChildren={<FormattedMessage id={'false'} />}
+                />
+              ) : (
+                <ProFormText width="lg" name="value" label={<FormattedMessage id="value" />} />
+              )
+            }
+          </ProForm.Item>
```

Comment:
looks a little __hacky__ 
imho it would be better to create a specific component for this 

`SettingsValue` or something like that